components/brave_wallet/browser/json_rpc_service_unittest.cc

```diff
@@ -2458,4 +2476,61 @@ TEST_F(JsonRpcServiceUnitTest, GetSolanaAccountInfo) {
                            l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
 }
 
+TEST_F(JsonRpcServiceUnitTest, GetSolanaFeeForMessage) {
+  std::string json = R"(
+    {
+      "jsonrpc":"2.0","id":1,
+      "result": {
+        "context":{"slot":123065869},
+        "value": 12345
+      }
+    }
+  )";
+  SetInterceptor("getFeeForMessage", "", json);
+  std::string base64_encoded_string;
+  base::Base64Encode("test", &base64_encoded_string);
+
+  TestGetSolanaFeeForMessage(base64_encoded_string, 12345,
+                             mojom::SolanaProviderError::kSuccess, "");
+  std::string base58_encoded_string = "JvSKSz9YHfqEQ8j";
+  // Message has to be base64 encoded string and non-empty.
+  TestGetSolanaFeeForMessage(
+      "", 0, mojom::SolanaProviderError::kInvalidParams,
+      l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+  TestGetSolanaFeeForMessage(
+      base58_encoded_string, 0, mojom::SolanaProviderError::kInvalidParams,
+      l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+
+  // value can be null for an account not on chain.
+  SetInterceptor("getFeeForMessage", "",
+                 R"({
+                      "jsonrpc":"2.0",
+                      "result":{
+                      "context":{"slot":123121238},"value":null},"id":1
+                 })");
+  TestGetSolanaFeeForMessage(base64_encoded_string, 0,
+                             mojom::SolanaProviderError::kSuccess, "");
+
+  // Response parsing error
+  SetInterceptor("getFeeForMessage", "",
+                 "{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":\"0\"}");
+  TestGetSolanaFeeForMessage(
+      base64_encoded_string, 0, mojom::SolanaProviderError::kParsingError,
+      l10n_util::GetStringUTF8(IDS_WALLET_PARSING_ERROR));
+
+  // JSON RPC error
+  SetInterceptor("getFeeForMessage", "",
+                 "{\"jsonrpc\":\"2.0\",\"id\":1,\"error\":"
```

Comment:
nit: I think it's preferred to have raw string literals for JSON.

Reply:
Fixed by interactive rebase in https://github.com/brave/brave-core/pull/12800/commits/6dc824bba57e7b1028a7463ab9b69b6b6032571c